src/armado/sqlite_index.py

```diff
@@ -254,6 +374,24 @@ def get_doc(self, docid):
             row[0] = to_filename(row[1])
         return row
 
+    def search(self, keys):
+        """Returns all the values that are found for those keys.
+
+        The AND boolean operation is applied to the keys.
+        """
+        keys = list(map(cdpindex.normalize_words, keys))
+        files_yielded = set()
+        docset = Search(self.db, keys)
+        for score, ndoc in docset.ordered:
+            doc_data = self.get_doc(ndoc)
+            if not doc_data[0] in files_yielded:
```

Comment:
Please add a comment about what is `doc_data[0]` and why are you checking on that to exclude the result.